### PR TITLE
Fixed cayley build to pack website assets

### PR DESCRIPTION
**Pull request by iddan**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This is a second attempt. Incorporating comments from the previous pull request

### CODE REVIEW

File: Formula/cayley.rb [hunk 1]
```diff
@@ -16,12 +16,18 @@ class Cayley < Formula
   depends_on "mercurial" => :build
 
   def install
-    ENV["GOPATH"] = buildpath
+    ENV["GO111MODULES"] = "on"
```
Comment (SMillerDev) [inline Formula/cayley.rb:19]: This is default in the latest go, it shouldn't be needed

File: Formula/cayley.rb [hunk 2]
```diff
@@ -16,12 +16,18 @@ class Cayley < Formula
...
 
     dir = buildpath/"src/github.com/cayleygraph/cayley"
     dir.install buildpath.children
 
     cd dir do
+      # Install dependencies
+      system "go", "mod", "download"
```
Comment (SMillerDev) [inline Formula/cayley.rb:26]: https://docs.brew.sh/Acceptable-Formulae#we-dont-like-install-scripts-that-download-unversioned-things

File: Formula/cayley.rb [hunk 3]
```diff
@@ -84,6 +91,20 @@ def plist; <<~EOS
...
     assert_match version.to_s, shell_output("#{bin}/cayley version")
+
+    # Check UI
+    begin
+      pid = fork do
+        exec "#{bin}/cayley", "http"
+      end
+      sleep 3
+      response = shell_output("curl -s 127.0.0.1:64210")
+      refute_empty response, "Response must not be empty"
```
Comment (SMillerDev) [inline Formula/cayley.rb:104]: Please use assert_match and check for some output.

File: Formula/cayley.rb (hunk 2 above)
Comment (iddan) [reply Formula/cayley.rb:26]: Will wrap with `resource`

Comment (iddan) [reply Formula/cayley.rb]: Eventually no need for go mod download.

File: Formula/cayley.rb [hunk 4]
```diff
@@ -84,6 +86,20 @@ def plist; <<~EOS
...
     assert_match version.to_s, shell_output("#{bin}/cayley version")
+
+    # Check UI
+    begin
+      pid = fork do
+        exec "#{bin}/cayley", "http"
+      end
+      sleep 3
+      response = shell_output("curl -s -i 127.0.0.1:64210")
+      assert_match %r{/HTTP\/1.1 200 OK/}, response
```
Comment (SMillerDev) [inline Formula/cayley.rb:99]: This can just be a normal string.

File: Formula/cayley.rb [hunk 5]
```diff
@@ -84,6 +86,20 @@ def plist; <<~EOS
...
+    ensure
```
Comment (SMillerDev) [inline Formula/cayley.rb:100]: This part shouldn't be needed anymore.

File: Formula/cayley.rb (hunk 4 above)
Comment (iddan) [reply Formula/cayley.rb:99]: Maybe I miss something, but this is only one line in the output.

File: Formula/cayley.rb (hunk 5 above)
Comment (iddan) [reply Formula/cayley.rb:100]: Will remove

File: Formula/cayley.rb (hunk 4 above)
Comment (SMillerDev) [reply Formula/cayley.rb:99]: Yeah, but assert_match will also work if you feed it a string to match. And in this case that's possible and easier to read.

File: Formula/cayley.rb [hunk 6]
```diff
@@ -84,6 +86,15 @@ def plist; <<~EOS
   end
 
   test do
+    # Check version
```
Comment (SMillerDev) [inline Formula/cayley.rb:89]: No need to comment this

File: Formula/cayley.rb [hunk 7]
```diff
@@ -84,6 +86,15 @@ def plist; <<~EOS
...
     assert_match version.to_s, shell_output("#{bin}/cayley version")
+
+    # Check UI
```
Comment (SMillerDev) [inline Formula/cayley.rb:92]: Same here

File: Formula/cayley.rb [hunk 8]
```diff
@@ -84,6 +86,15 @@ def plist; <<~EOS
...
+    fork do
+      exec "#{bin}/cayley", "http"
```
Comment (SMillerDev) [inline Formula/cayley.rb:94]: if you can specify the port, please do using `free_port` like https://github.com/Homebrew/homebrew-core/blob/master/Formula/traefik.rb#L61

**Comment by iddan** [comment]
@SMillerDev any other comments? If not, can you approve please?

**Comment by SMillerDev** [comment]
Can you squash everything into one `cayley: pack website assets` commit?

**Comment by iddan** [comment]
Done

**Comment by SMillerDev** [comment]
Thanks @iddan  Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**